Mathlib/Analysis/Normed/Field/ProperSpace.lean

```diff
@@ -0,0 +1,37 @@
+/-
+Copyright (c) 2024 Yakov Pechersky. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yakov Pechersky
+-/
+
+import Mathlib.Analysis.Normed.Field.Lemmas
+import Mathlib.Analysis.SpecificLimits.Basic
+import Mathlib.Topology.MetricSpace.ProperSpace
+
+/-!
+# Proper nontrivally normed fields
+
+This file contains a direct result of normed fields that are `ProperSpace`s.
+
+## Main results
+
+* `ProperSpace.of_weaklyLocallyCompactSpace_of_nontriviallyNormedField`
+-/
+
+open Metric Filter
+
+/-- A weakly locally compact normed field is proper. -/
+lemma ProperSpace.of_weaklyLocallyCompactSpace_of_nontriviallyNormedField
+    (𝕜 : Type*) [NontriviallyNormedField 𝕜] [WeaklyLocallyCompactSpace 𝕜] :
+    ProperSpace 𝕜 := by
+  rcases exists_isCompact_closedBall (0 : 𝕜) with ⟨r, rpos, hr⟩
+  rcases NormedField.exists_one_lt_norm 𝕜 with ⟨c, hc⟩
+  have hC : ∀ n, IsCompact (closedBall (0 : 𝕜) (‖c‖^n * r)) := fun n ↦ by
+    have : c ^ n ≠ 0 := pow_ne_zero _ <| fun h ↦ by simp [h, zero_le_one.not_lt] at hc
+    convert hr.smul (c ^ n)
+    ext
```

Comment:
At this point the goal is `closedBall 0 (‖c‖ ^ n * r) = c ^ n • closedBall 0 r`. Generalising to replace `c ^ n` with an arbitrary element of `𝕜`, this looks like a lemma worth having in its own right (& maybe for normed vector spaces over normed fields, or even normed modules over normed rings, not just for normed fields themselves)

Reply:
We do have the more general result, but it lives in NormedSpace.Pointwise which I was trying to avoid importing.